specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/workspaceRP.json

```diff
@@ -3522,7 +3602,7 @@
           "type": "string"
         },
         "managedNetwork": {
-          "$ref": "#/definitions/ManagedNetworkSettings"
+          "$ref": "#/definitions/ManagedNetworkSettingsEx"
```

Comment:
>ManagedNetworkSettingsEx [](http://example.com/codeflow?start=33&length=24)

I'm not clear why you are doing it this way, as opposed to just adding properties on ManagedNetworkSettings. Want to explain?

Reply:
sorry take back what I just said, this is intended, I agree that it can be achieved through adding this property as optional property in ManagedNetworkSettings but this swagger is generated based on the code, so if this also will work I hope we can keep it so we don't have to manually fix the swagger after autogeneration.

Reply:
Sorry, but this just doesn't seem like good API design to me. 

As guiding principles, we should be striving to make the latest version of our API look simple and well designed -- with possible considerations made for backcompatibility. For the design here, I'm not even convinced helps with back compatibility. E.g. for SDKs, isn't it going to be *less* backwards compatible, because people will have to replace references to ManagedNetworkSettings with ManagedNetworkSettingsEx - they can't just update the package version and recompile, right?

I would like to ask if you can instead find workarounds for this in the implementation side, to do the appropriate validations, population, serialization etc with just one swagger payload Definition, ManagedNetworkSettings, not two (where one of them might never really be used with this api version).

Reply:
OK good point, I will add the property in ManagedNetworkSettings directly, I think I can override the swagger auto-generation logic for this.

Reply:
Updated, thanks for the good feedback~